File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.cpp
bool compare(vector<int>v1,vector<int>v2){
            return v1[0]<v2[0]?true:(v1[0]==v2[0]?(v1[1]<v2[1]?true:false):false);
}


class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        
            sort(points.begin(),points.end(),compare);
            vector<vector<int>>ans;
            //ans.push_back(points[0]);
            for(int i=0;i<points.size();i++){
                    if(ans.size()!=0 and points[i][0]<=ans.back()[1]){
                                if(points[i][1]>ans.back()[1]){
                                    int temp=ans.back()[1];
                                    ans.pop_back();
                                    ans.push_back({points[i][0],temp});
                                }
                                else{
                                        ans.pop_back();
                                        ans.push_back({points[i]});
                                }
                    }
                    else{
                        
                                ans.push_back(points[i]);
                        
                    }
                    
            }
        
                return ans.size();
        }
    
};

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    
    
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        
        
        new_points=sorted(points,key=lambda x:x[0])
        ans=[new_points[0]]
        i=1
        while(i<len(new_points)):
            
        
            if new_points[i][0]>ans[-1][1]:
                    ans.append(new_points[i])
            else:
                    val=ans.pop(-1)
                    
                    ans.append([max(val[0],new_points[i][0]),min(val[1],new_points[i][1])])

            i+=1
           
        print(ans)
        return len(ans)
            